api/src/services/payload.ts

```diff
@@ -628,8 +628,8 @@ export class PayloadService {
 						// http route, and or a bigInteger in the DB
 						if (
 							isNil(existingRecord[relation.field]) === false &&
-							(existingRecord[relation.field] == parent ||
-								existingRecord[relation.field] == payload[currentPrimaryKeyField])
+							(existingRecord[relation.field] === parent ||
+								existingRecord[relation.field] === payload[currentPrimaryKeyField])
```

Comment:
We may need to further confirm whether this is ok, since the comment above (particularly line 626) does state the following:

> NOTE: We use == here, as the primary key might be reported as a string instead of number, coming from the http route, and or a bigInteger in the DB

Reply:
Great catch! Didn't spot that comment. In this case we can, for example, disable the ESLint rule with a clear statement that this is intentional.